experiments/deep_learning_with_pytorch/mlp_parameter_tuning.py

```diff
@@ -0,0 +1,34 @@
+from pathlib import Path
+from experiments.deep_learning_with_pytorch.surnames import *
+
+from transfer_nlp.plugins.config import register_plugin, ExperimentConfig
+from transfer_nlp.plugins.reporters import ReporterABC
+from transfer_nlp.runner.experiment_runner import ExperimentRunner
+
+
+@register_plugin
+class MyReporter(ReporterABC):
+    def __init__(self):
+        self.reported = False
+
+    def report(self, name: str, experiment: ExperimentConfig, report_dir: Path):
+
+        with open(report_dir / 'metrics_report.txt', 'w') as reporting:
+            reporting.write(f"Metrics reporting for experiment {name}\n")
+            reporting.write("#"*50 + '\n')
+
+            for mode, metrics in experiment['trainer'].metrics_history.items():
+                reporting.write(f"Reporting metrics in {mode} mode\n")
+                for metric, values in metrics.items():
+                    reporting.write(f"{metric}: [{', '.join([str(value) for value in values])}]\n")
+
+
+if __name__ == "__main__":
+
+    dir = Path(__file__).parent
+    home_env = str(Path.home() / 'work/transfer-nlp-data')
+    ExperimentRunner.run_all(experiment=dir / 'mlp_parameter_tuning.json',
+                             experiment_config=dir / 'mlp_parameter_tuning.cfg',
+                             report_dir=dir / 'reports',
```

Comment:
probably not great to create directories in the repo area that aren't meant to be checked in. i would suggest using something under `home_env` instead. also, i think we should use a timestamp dir under reports to indicate when the code ran as best practice. so finally i would suggest something like
`home_env / 'mlp_parameter_fine_tuning' / 20190521` as the report dir where 20190521 is the current date.

Reply:
also probably don't use `dir` as a variable name: https://docs.python.org/3/library/functions.html#dir